packages/workbox-sw/test/sw/directory-index-custom.js

```diff
@@ -59,4 +59,45 @@ describe(`Test Directory Index`, function() {
       self.dispatchEvent(fetchEvent);
     });
   });
+
+  it(`should take the ignoreUrlParametersMatching setting into account when using the directoryIndex`, function() {
+    const urlParameter = 'test';
+    const ignoreUrlParametersMatching = [new RegExp(urlParameter)];
+    const baseUrl = '/example/url/';
+    const directoryIndex = 'custom.html';
+
+    const urlToPrecache = baseUrl + directoryIndex;
+    const urlToRequest = baseUrl + '?' + urlParameter;
+
+    // We need to create a stub for match() that returns a real Promise,
+    // rather than null, or else Firefox is unhappy.
+    const stub = sinon.stub(Cache.prototype, 'match')
+      .callsFake(() => Promise.resolve(null));
+    stubs.push(stub);
+
+    const workboxSW = new WorkboxSW({
+      directoryIndex,
+      ignoreUrlParametersMatching,
+    });
+    workboxSW.precache([urlToPrecache]);
+
+    return new Promise((resolve, reject) => {
+      const fetchEvent = new FetchEvent('fetch', {
+        // Request the URL without the directory index, but with the URL param.
+        request: new Request(urlToRequest),
+      });
+      fetchEvent.respondWith = (promiseChain) => {
+        promiseChain.then(() => {
+          // Make sure that the cache.match() stub was called, indicating that
+          // there was a match against the precached URL.
+          if (stub.called) {
```

Comment:
What is caches.match called with? Should it be with or without the directoryIndex value or not?

Reply:
Added in a check.